rosbag_snapshot/src/snapshotter.cpp

```diff
@@ -161,9 +195,14 @@ bool MessageQueue::preparePush(int32_t size, ros::Time const& time)
     while (queue_.size() != 0 && size_ + size > options_.memory_limit_)
       _pop();
 
-  // If duration limit is encforced, remove elements from front of queue until duration limit would be met once message
+  // If topic is latched, ignore duration limit but remove all previous messages from the specified callerid
+  if (_is_latched())
+  {
+    _remove_callerid(callerid);
```

Comment:
`getCallerId()` can return an empty string, and that gets passed into this method. Should we check if `callerid` is the empty string here? I guess it can't hurt to try to remove it anyway.

Reply:
If it's an empty string it would mean the publisher is sending messages with that field missing in the connection header (not sure if that's possible) or the field is there but is somehow empty (not sure if that's possible either). In that case it will match other messages that also return the same empty string which could be a problem if there is more than one publisher doing that, but if that's somehow happening then it would kind of break this whole thing anyway.